app/components/projectAssignment/workWeekInput.tsx

```diff
@@ -59,45 +78,85 @@ export const WorkWeekInput = ({
 		if (values.actualHours !== "") {
 			variables.actHours = parseInt(values.actualHours);
 		}
+
 		upsertWorkweek({
 			variables
-		}).then((res) => {
-			refetchUserList();
-			refetchProjectList();
-		});
+		})
 	};
+
+	const onFillForwardClick = async (inputName: string,
+		targetCweek: number, values: FormikValues) => {
+		const variables: FillForwardVariablesType = {
+			assignmentId: values.assignmentId,
+		};
+
+		if ((inputName === ESTIMATED_HOURS && values.estimatedHours === '') ||
+			(inputName === ACTUAL_HOURS && values.actualHours === '')) {
+			return;
+		}
+
+		const weekNumbersPerScreen = getWeekNumbersPerScreen(months);
+
+		let filteredWeeks = filterWeeksForFillForward(weekNumbersPerScreen, targetCweek, inputName);
+		if (assignment.endsOn || assignment.startsOn) {
+			filteredWeeks = filteredWeeks.filter(week =>
+				assignmentContainsCWeek(assignment, week.cweek, week.year))
+		}
+
+		if (inputName === ESTIMATED_HOURS && values.estimatedHours !== "") {
+			variables.workWeeks = filteredWeeks.map(week => ({
+				cweek: week.cweek,
+				estimatedHours: parseInt(values.estimatedHours),
+				year: week.year
+			}));
+		}
+
+		if (inputName === ACTUAL_HOURS && values.actualHours !== "") {
```

Comment:
@fermion I am already filtered future hours for actual hours it in function above

Reply:
@fermion so future hours for actual hours will not be send

Reply:
🆗 that's great! I just wanted to mention in case there was confusion on the response not having the values that were passed to the backend. As long as the UI uses the values on the response for rendering we should be good 👍🏻